pkg/xds/server/callbacks/dataplane_status_tracker.go

```diff
@@ -116,13 +116,15 @@ func (c *dataplaneStatusTracker) OnStreamRequest(streamID int64, req util_xds.Di
 
 	// infer Dataplane id
 	if state.dataplaneId == (core_model.ResourceKey{}) {
-		if id, err := core_xds.ParseProxyIdFromString(req.NodeId()); err == nil {
-			state.dataplaneId = core_model.ResourceKey{Mesh: id.Mesh, Name: id.Name}
+		if proxyId, err := core_xds.ParseProxyIdFromString(req.NodeId()); err == nil {
+			state.dataplaneId = proxyId.ToResourceKey()
 			if err := readVersion(req.Metadata(), state.subscription.Version); err != nil {
 				statusTrackerLog.Error(err, "failed to extract version out of the Envoy metadata", "streamid", streamID, "metadata", req.Metadata())
 			}
 			// kick off async Dataplane status flusher
-			go c.createStatusSink(state).Start(state.stop)
+			if proxyId.Type() != mesh_proto.IngressDpType {
```

Comment:
put a comment that we don't support insights for ZoneIngress

Reply:
added support for insights